library/core/src/primitive_docs.rs

```diff
@@ -731,12 +731,59 @@ mod prim_array {}
 /// assert_eq!(2 * pointer_size, std::mem::size_of::<Box<[u8]>>());
 /// assert_eq!(2 * pointer_size, std::mem::size_of::<Rc<[u8]>>());
 /// ```
+///
+/// ## Trait Implementations
+///
+/// There are several implementations of common traits for slices. Some examples
+/// include:
+///
+/// * [`Clone`]
+/// * [`Eq`], [`Ord`] - for slices whose element type are [`Eq`] or [`Ord`].
+/// * [`Hash`] - for slices whose element type is [`Hash`].
+///
+/// ## Iteration
+///
+/// The slices implement `IntoIterator`. The iterator yields references to the
+/// slice elements.
+///
+/// ```
+/// let numbers = &[0, 1, 2];
+/// for n in numbers {
+///     println!("{} is a number!", n);
+/// }
+/// ```
+///
+/// The mutable slice yields mutable references to the elements:
+///
+/// ```
+/// let mut scores = [7, 8, 9];
+/// for score in &mut scores[..] {
+///     *score += 1;
+/// }
+/// ```
+///
+/// This iterator yields mutable references to the slice's elements, so while
+/// the element type of the slice is `i32`, the element type of the iterator is
+/// `&mut i32`.
+///
+/// * [`.iter`] and [`.iter_mut`] are the explicit methods to return the default
+///   iterators.
+/// * Further methods that return iterators are [`.split`], [`.splitn`],
+///   [`.chunks`], [`.windows`] and more.
+///
+/// [`Hash`]: core::hash::Hash
+/// [`.iter`]: slice::iter
+/// [`.iter_mut`]: slice::iter_mut
+/// [`.split`]: slice::split
+/// [`.splitn`]: slice::splitn
+/// [`.chunks`]: slice::chunks
+/// [`.windows`]: slice::windows
 #[stable(feature = "rust1", since = "1.0.0")]
 mod prim_slice {}
 
 #[doc(primitive = "str")]
 //
-/// String slices.
+/// Unicode string slices.
```

Comment:
I'd be careful with using the term 'unicode string'. That's used in some other languages/on other platforms to refer to `[u32]` or `[u16]` string representations. You could mention UTF-8 specifically, but that's already mentioned below and might be a bit too much detail for the first line.

Reply:
I could just undo this change. Also is "slices" a bit over-technical or confusing? I'm not how sure "slice" applies to a `const &'static str` for example. How about "The primitive string type"?

Reply:
"Slices" might be important to not cause confusion with the `String` type. 

> I'm not how sure "slice" applies to a `const &'static str` for example.

That's still a pointer+size to a 'slice' of the program (where the string is stored). But I suppose the main point is that if you take any substring, you still have the same type.

Reply:
I'll just revert this.